core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualOutputWriter.scala

```diff
@@ -831,6 +861,93 @@ object QualOutputWriter {
     }
   }
 
+  def constructUnsupportedOperatorsInfo(
+      sumInfo: QualificationSummaryInfo,
+      headersAndSizes: LinkedHashMap[String, Int],
+      delimiter: String = TEXT_DELIMITER,
+      prettyPrint: Boolean,
+      reformatCSV: Boolean = true): Seq[String] = {
+    val reformatCSVFunc: String => String =
+      if (reformatCSV) str => StringUtils.reformatCSVString(str) else str => stringIfempty(str)
+    val appId = sumInfo.appId
+    val readFormat = sumInfo.readFileFormatAndTypesNotSupported
+    val writeFormat = sumInfo.writeDataFormat
+    val unsupportedExecs = sumInfo.unSupportedExecs
+    val unsupportedExprs = sumInfo.unSupportedExprs
+    val unsupportedExecExprsMap = sumInfo.unsupportedExecstoExprsMap
+    val unsupportedOperatorsOutputRows = new ArrayBuffer[String]()
+
+    if (readFormat.nonEmpty) {
+      val unsupportedReadFormatRows = readFormat.map { format =>
+        val readFormatAndType = format.split("\\[")
+        val readFormat = readFormatAndType(0)
+        val readType = if (readFormatAndType.size > 1) {
+          s"Types not supported - ${readFormatAndType(1).replace("]", "")}"
+        } else {
+          ""
+        }
+        val data = ListBuffer(
+          reformatCSVFunc(appId) -> headersAndSizes(APP_ID_STR),
+          "Read" -> headersAndSizes(UNSUPPORTED_TYPE),
```

Comment:
AFAIR, Ryan used `recormatCSVFunc` with all string cells in other CSV output files. but in this CSV file, we do it only for the AppID column.
Shouldn't we do the same here to be consistent?  Especially, that "Notes" can have some special characters that break the CSV file. 
For example, in the below code, `exprs` is a variable that could have "`;`, `,`..etc".

```
          s"$exec Exec is not supported as expressions are not supported -  `${exprs}`" ->
            headersAndSizes(NOTES)
```

And I can see that the final CSV file does not guarantee that the file is protected from special character.

Also, Empty column should be `""` but the `Notes` in the CSV file is an empty string

```
App ID,Unsupported Type,Details,Notes
"app-20210509200722-0001",Exec,BroadcastHashJoin,
"app-20210509200722-0001",Exec,SortMergeJoin,
"app-20210509200722-0001",Exec,CollectLimit,
"app-20210509200722-0001",Exec,Execute InsertIntoHadoopFsRelationCommand json,
"app-20210509200722-0001",Exec,SetCatalogAndNamespace,
"app-20210509200722-0001",Exec,LocalTableScan,
"app-20210509200722-0001",Exec,Subquery,
"app-20210509200722-0001",Write,json,
```

Reply:
Thanks for pointing this. Updated it for all columns.